macros/utils/data_types/cast_column.sql

```diff
@@ -43,6 +55,15 @@
     cast({{ elementary.edr_cast_as_timestamp(timestamp_field) }} as {{ elementary.edr_type_date() }})
 {%- endmacro -%}
 
+{# Athena needs explicit conversion for ISO8601 timestamps used in buckets_cte #}
+{%- macro athena__edr_cast_as_date(timestamp_field) -%}
+    coalesce(
+        try_cast({{ timestamp_field }} as {{ elementary.edr_type_date() }}),
```

Comment:
Why do we need `try_cast` and not normal `cast`? try_cast guarantee to work in all the cases, but can lead to some data lost as if the casting fail NULL will be returned.

Reply:
We need it to work for different timestamp formats (standard SQL ts literals and ISO8601). `cast()` doesn't return NULL for failed cases, at least not for timestamps. E.g.
```
select coalesce(cast('2023-10-01T10:10:10' as timestamp), from_iso8601_timestamp('2023-10-01T10:10:10')) ts
```
will fail, but
```
select coalesce(try_cast('2023-10-01T10:10:10' as timestamp), from_iso8601_timestamp('2023-10-01T10:10:10')) ts
```
will work as expected

Reply:
actually, that works because the first try_cast return NULL, then it uses the 2nd value. Thanks for clarifying it, make sense.